.github/workflows/tests.yml

```diff
@@ -41,30 +41,39 @@ jobs:
       run: python scripts/lint.py rules/
 
   tests:
-    name: Tests in ${{ matrix.python }}
-    runs-on: ubuntu-latest
+    name: Tests in ${{ matrix.python-version }} on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
     needs: [code_style, rule_linter]
     strategy:
       fail-fast: false
       matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        # across all operating systems
+        python-version: [2.7, 3.9.1]
```

Comment:
I think this should be 3.9 instead of 3.9.1. According to setup-python documentation, using patch versions will take longer if the exact version is not already installed in the runner due to more recent version. Specifying only a major and minor version uses the most recent patch version, which is the one that we IMO should use and it is faster.

Reply:
I would include 3.6 here, either now or when we get rid of Python 2.7